contracts/utils/MultiRateLimited.sol

```diff
@@ -0,0 +1,442 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+pragma solidity ^0.8.4;
+
+import {CoreRef} from "../refs/CoreRef.sol";
+import {TribeRoles} from "./../core/TribeRoles.sol";
+import {RateLimited} from "./RateLimited.sol";
+import {IMultiRateLimited} from "./IMultiRateLimited.sol";
+import {Math} from "@openzeppelin/contracts/utils/math/Math.sol";
+import {SafeCast} from "@openzeppelin/contracts/utils/math/SafeCast.sol";
+
+/// @title contract for putting a rate limit on how fast an address can perform an action e.g. Minting
+/// there are two buffers, one buffer which is each individual addresses's current buffer,
+/// and then there is a global buffer which is the buffer that each individual address must respect as well
+/// @author Fei Protocol
+contract MultiRateLimited is RateLimited, IMultiRateLimited {
+    using SafeCast for *;
+
+    /// @notice the struct containing all information per rate limited address
+    struct RateLimitData {
+        uint32 lastBufferUsedTime;
+        uint144 bufferCap;
+        uint112 rateLimitPerSecond;
+        uint144 bufferStored;
+    }
+
+    /// @notice rate limited address information
+    mapping(address => RateLimitData) public rateLimitPerAddress;
+
+    /// @notice max rate limit per second allowable by non governor per contract
+    uint256 public maxRateLimitPerSecond;
+
+    /// @notice max buffer cap allowable by non governor per contract
+    uint256 public maxBufferCap;
+
+    /// @param coreAddress address of the core contract
+    /// @param _maxRateLimitPerSecond maximum amount of fei that can replenish per second ever, this amount cannot be changed by governance
+    /// @param _rateLimitPerSecond maximum rate limit per second per address
+    /// @param _maxRateLimitPerSecondMRL maximum rate limit per second per address in multi rate limited
+    /// @param _maxBufferCap maximum buffer cap in multi rate limited
```

Comment:
`individualMaxBufferCap` is clearer to me

Reply:
on global param too

Reply:
done